orion/pkg/repos/achieve_repo.go

```diff
@@ -212,35 +212,45 @@ func (ar *achieveRepo) Update(id uint, achieve domains.Achieve) error {
 	return handleSqlExecResult(execResult, 1, "achievement was not updated")
 }
 
-func (ar *achieveRepo) Delete(id uint) error {
-	statement := "DELETE FROM achievements WHERE id=?"
-	stmt, err := ar.db.Prepare(statement)
+func (ar *achieveRepo) Publish(ids []uint) error {
+	errorList := make([]domains.PublishErrorBody, 0)
+
+	// Begin Transaction
+	tx, err := ar.db.Begin()
+	str := "UPDATE achievements SET published_at=? WHERE id=? AND published_at IS NULL"
+	stmt, err := tx.Prepare(str)
 	if err != nil {
 		return err
 	}
 	defer stmt.Close()
 
-	execResult, err := stmt.Exec(id)
-	if err != nil {
-		return err
+	now := time.Now().UTC()
+	for _, id := range ids {
+		_, err := stmt.Exec(now, id)
+		if err != nil {
+			errorList = append(errorList, domains.PublishErrorBody{RowId: id, Error: err})
+		}
 	}
-	return handleSqlExecResult(execResult, 1, "achievement was not deleted")
+
+	// End Transaction
+	tx.Commit()
+
+	return domains.ConcatErrors(errorList)
```

Comment:
Yeah this is what I was saying earlier on my PR. You technically don't need the struct PublishErrorBody because instead of pushing structs to a list and then concatenating at the end, you can just build the error string on the fly by concatenating whenever you encounter an error in the for loop.

Reply:
That's true